x/incentive/keeper/rewards_earn.go

```diff
@@ -179,7 +182,18 @@ func (k Keeper) collectDerivativeStakingRewards(ctx sdk.Context, collateralType
 		// otherwise there's no validator or delegation yet
 		rewards = nil
 	}
-	return sdk.NewDecCoinsFromCoins(rewards...)
+
+	// Bug with NewDecCoinsFromCoins when calling passing 0 amount Coin, see
+	// https://github.com/cosmos/cosmos-sdk/pull/12903
+	// Fix is in Cosmos-SDK v0.47.0
```

Comment:
Do we need to compensate for this bug in our other uses of `sdk.NewDecCoinsFromCoins`?

Reply:
I looked through our other uses, mostly in tests and with incentive `RewardsPerSecond` - I don't believe these coins would have a zero amount, but it isn't validated. I can add some extra validation to make sure the coins in `RewardsPerSecond` are non-zero